src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops.cpp

```diff
@@ -0,0 +1,1074 @@
+        return false;
+
+    if(!IsValidBlockwiseGemmXdlops(
+           ctx, GemmMPerBlock, GemmNPerBlock, GemmKPerBlock, GemmMPerWave, GemmNPerWave, GemmKPack))
+        return false;
+
+    bool valid = false;
+    // n should be multiple of GemmKBlocks
+    {
+        const auto n = ConvolutionContextInterpreter::GetBatchN(ctx);
+        if(n % GemmKBlocks != 0)
+            return false;
+    }
+    // check blockwise GEMM size
+    {
+        int gemm_m       = -1;
+        int gemm_n       = -1;
+        int gemm_k_total = -1;
+
+        std::tie(std::ignore, gemm_m, gemm_n, gemm_k_total) =
+            ConvHipImplicitGemmWrwV4R4Xdlops::CalculateGemmSize(ctx);
+
+        if(gemm_k_total % GemmKPack != 0)
+            return false;
+
+        const auto gemm_k = gemm_k_total / GemmKPack;
+
+        if(gemm_k % GemmKBlocks != 0)
+            return false;
+        const auto gemm_k_sub = gemm_k / GemmKBlocks;
+
+        if(!(gemm_m % GemmMPerBlock == 0 && gemm_n % GemmNPerBlock == 0 &&
+             gemm_k_sub % GemmKPerBlock == 0))
+            return false;
+    }
+
+    // check blockwise copy of A matrix
+    {
+        std::tie(std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, valid) =
+            CalculateGemmABlockCopyPerformanceParameters(ctx);
+
+        if(!valid)
+            return false;
+    }
+
+    // check blockwise copy of B matrix
+    {
+        std::tie(std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, valid) =
+            CalculateGemmBBlockCopyPerformanceParameters(ctx);
+
+        if(!valid)
+            return false;
+    }
+
+    // check LDS allocation
+    std::size_t lds_size = 0;
+    std::tie(lds_size, valid) = CalculateLdsNumberOfByte(ctx);
+
+    return (valid and lds_size <= get_lds_max_number_of_byte());
+}
+
+// Used by GenericSearch, not used by EuristicInit
+// Return false if a performance config is known to be sub-optimal, comparing to other performance
+// config inside tuning range
+bool PerformanceImplicitGemmWrwV4R4Xdlops::IsFastToBeUsedForTuning(
+    const ConvolutionContext& ctx) const
+{
+    // somehow, 128x128 wave-wise GEMM tend to spill register
+    // TODO revisit this when 128x128 wave-wise GEMM become efficient
+    {
+        if(GemmMPerWave * GemmNPerWave > 64 * 128)
+            return false;
+    }
+
+#if WORKAROUND_SWDEV_240356
+    {
+        if(ctx.IsBfp16() && GemmMPerWave * GemmNPerWave > 64 * 64)
+            return false;
+    }
+#endif
+
+    // don't need too many blocks
+    {
+        int gemm_m = 0;
+        int gemm_n = 0;
+
+        std::tie(std::ignore, gemm_m, gemm_n, std::ignore) =
+            ConvHipImplicitGemmWrwV4R4Xdlops::CalculateGemmSize(ctx);
+
+        // this is grid size using current blockwise-GEMM
+        const int grid_size = (gemm_m * gemm_n) / (GemmMPerBlock * GemmNPerBlock);
+
+        // this the the biggest blockwise-GEMM you can do
+        int max_blockwise_gemm_size =
+#if WORKAROUND_SWDEV_240356
```

Comment:
Sam as above. Do you need this workaround?

Reply:
please remove this workaround. The issue should have been resolved by #403

Reply:
OKay,I will remove it and test today